chromium_src/chrome/browser/ui/hats/hats_service_browsertest.cc

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <vector>
+
+#include "base/test/scoped_feature_list.h"
+#include "base/time/time.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/metrics/chrome_metrics_service_accessor.h"
+#include "chrome/browser/profiles/profile_impl.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/hats/hats_service.h"
+#include "chrome/browser/ui/hats/hats_service_factory.h"
+#include "chrome/common/chrome_features.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/metrics_services_manager/metrics_services_manager.h"
+#include "content/public/test/browser_test.h"
+#include "net/dns/mock_host_resolver.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+// Simplified version of the upstream tests modified to reflect the change in
+// the HatsService chromium_src override.
+
+namespace {
+
+base::test::ScopedFeatureList::FeatureAndParams settings_probability_one{
+    features::kHappinessTrackingSurveysForDesktopSettings,
+    {{"probability", "1.000"},
+     {"survey", kHatsSurveyTriggerSettings},
+     {"en_site_id", "test_site_id"}}};
+
+class ScopedSetMetricsConsent {
+ public:
+  // Enables or disables metrics consent based off of |consent|.
+  explicit ScopedSetMetricsConsent(bool consent) : consent_(consent) {
+    ChromeMetricsServiceAccessor::SetMetricsAndCrashReportingForTesting(
+        &consent_);
+  }
+
+  ScopedSetMetricsConsent(const ScopedSetMetricsConsent&) = delete;
+  ScopedSetMetricsConsent& operator=(const ScopedSetMetricsConsent&) = delete;
+
+  ~ScopedSetMetricsConsent() {
+    ChromeMetricsServiceAccessor::SetMetricsAndCrashReportingForTesting(
+        nullptr);
+  }
+
+ private:
+  const bool consent_;
+};
+
```

Comment:
nit: anonymous namespace should end here.

Reply:
Oddly that's how it was upstream. Moved in f34bdb3.